### DISCUSSION TITLE
Stops with only 1 trial and reports it as the best trial!

### DISCUSSION THREAD

**Discussion by VidyaKamath1089** [optuna/optuna#5159] [Q&A] [answered]
### Expected behavior

I have given n_trials = 20 and timeout= 600 in study. optimize()
Yet after one trial is complete the study.optimize terminates and reports trial 0 as best trial. 
Why is it stopping after one trial ? 
Note : I gave print statement "Study optimized.. " after study.optimize() .

### Environment

- Optuna version: 3.6.0.dev
- Python version: 3.11.5
- OS:Linux-5.15.0-91-generic-x86_64-with-glibc2.35
- (Optional) Other libraries and their versions: 
- PyTorch Version: 2.1.2+cu118
- cudatoolkit:  11.8.0 

### Error messages, stack traces, or logs

```shell
[I 2023-12-21 16:12:48,750] Trial 0 finished with value: 7397.056396484375 and parameters: {'optimizer': 'SGD', 'lr': 1.0132448516734057e-08}. Best is trial 0 with value: 7397.056396484375.
Study optimized.. 
Extracting Pruned and complete Trials...
Study statistics:
  Number of finished trials:  1
  Number of pruned trials:  0
  Number of complete trials:  1
Best trial:
  Value:  7397.056396484375
  Params:
    optimizer: SGD
    lr: 1.0132448516734057e-08
```

### Steps to reproduce

1. create model
2. create dataloaders (train and val) 
3. create any objective function

```python
# python code
study = optuna.study.study.create_study(direction="minimize") 
''' optuna.create_study and optuna.study.create_study NOT FOUND!  So using optuna.study.study.create_study to access create_study function in study.py file. '''

study.optimize(objective, n_trials=10, timeout=600)

# returns here just after completing a single trial!!! 
print("Study optimized.. \n  Extracting Pruned and complete Trials... ")

print("Study statistics: ")
print("  Number of finished trials: ", len(study.trials))
print("  Number of pruned trials: ", len(pruned_trials))
print("  Number of complete trials: ", len(complete_trials))

print("Best trial:")
trial = study.best_trial

print("  Value: ", trial.value)

print("  Params: ")
for key, value in trial.params.items():
     print("    {}: {}".format(key, value))
```

### Additional context (optional)

I am using optuna to perform hyperparameter tuning for an object detection model trained from scratch on pascal voc dataset. Model is created using torch.nn and includes batchnorm and relu layers. using SSD for detection.

**Comment by nzw0301** [comment, marked as answer]
> Why is it stopping after one trial ?

I think this is because your first trial takes more than 600 sec as specified by `timeout`.

**Reply by VidyaKamath1089** [reply]
I removed timeout and it works fine..

**Reply by nzw0301** [reply]
Thank you for trying my suggestion and share the result!

**Comment by VidyaKamath1089** [comment]
Thank you very much. I will try to run again without specifying timeout.
